test/unit/SingleSignatureDirtyHack.t.sol

```diff
@@ -0,0 +1,386 @@
+                bytes1(0x02),
+                LibRLP.p(block.chainid).p(nonce).p(gasPriorityPrice).p(gasPrice).p(gasLimit).p(to).p(0 wei).p(data).p(
+                    accessList.encode()
+                ).encode()
+            )
+        );
+    }
+
+    function _transferEIP1559(
+        address operator,
+        SingleSignatureDirtyHack.TransferParams memory transferParams,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        PackedSignature memory sig,
+        bytes memory encodedAccessList
+    ) internal {
+        (uint256 gasPriorityPrice, AccessListElem[] memory accessList) =
+            abi.decode(encodedAccessList, (uint256, AccessListElem[]));
+        vm.prank(operator);
+        harness.transferFrom(
+            WITNESS_TYPESTRING_SUFFIX, transferParams, gasPriorityPrice, gasPrice, gasLimit, accessList, sig
+        );
+    }
+
+    function testEIP1559Transfer(
+        uint256 signerPk,
+        uint256 gasPriorityPrice,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce,
+        uint256 requestedAmount,
+        bytes32 salt,
+        address operator,
+        address accessedAcount,
+        bytes32[3] memory slots
+    ) public {
+        bytes memory extraData;
+        {
+            AccessListElem[] memory accessList = new AccessListElem[](1);
+            accessList[0] = AccessListElem({account: accessedAcount, slots: new bytes32[](3)});
+            accessList[0].slots[0] = slots[0];
+            accessList[0].slots[1] = slots[1];
+            accessList[0].slots[2] = slots[2];
+            extraData = abi.encode(gasPriorityPrice, accessList);
+        }
+
+        _commonTest(
+            signerPk,
+            gasPrice,
+            gasLimit,
+            to,
+            amount,
+            deadline,
+            nonce,
+            requestedAmount,
+            salt,
+            operator,
+            extraData,
+            _eip1559Payload,
+            _transferEIP1559
+        );
+    }
+
+    function _commonTest(
+        uint256 signerPk,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce,
+        uint256 requestedAmount,
+        bytes32 salt,
+        address operator,
+        bytes memory extraData,
+        function(uint256, uint256, uint256, address, bytes memory, bytes memory) view returns (bytes32) getPayload,
+        function(address, SingleSignatureDirtyHack.TransferParams memory, uint256, uint256, PackedSignature memory, bytes memory) internal
+            transfer
+    ) internal {
+        signerPk = bound(signerPk, 2, 115792089237316195423570985008687907852837564279074904382605163141518161494336);
+        gasLimit = bound(gasLimit, 0, 30_000_000);
+        deadline = bound(deadline, block.timestamp, type(uint256).max - 1);
+        nonce = bound(nonce, 1, type(uint64).max - 3);
```

Comment:
why is `nonce == 0` outside the scope of the test? and isn't the maximum nonce `type(uin64).max - 1`?

Reply:
0 is outside because it is not allowed in the current implementation
<img width="701" alt="image" src="https://github.com/user-attachments/assets/78f3778b-16a7-40ca-b382-45396da2b1f4" />

As for the maximum it can be `type(uin64).max - 2` as I increment the nonce to times in the tests. Changing that.

Reply:
oh! nice catch. :thinking: I wonder how we can solve this. maybe we need to store the next nonce, rather than the current nonce, so that the comparison can be `<` instead of `<=`. we definitely want a nonce of zero to be allowed

Reply:
yep just storing the `incomingNonce + 1` do the work, let me change it

Reply:
Also kept `type(uint64).max - 3` as max because I increment twice and `type(uint64).max` is not allowed